### PR TITLE
Fixes #3604, Fixes #3603 - Document ArgumentCompleter

### LINKED ISSUE #3603: Example is inadequate for Register-ArgumentCompleter.md
The sample scripts need to be explained more clearly. There is no documentation of the parameters passed to Register-ArgumentCompleter and no description of what the sample does that would allow a reader to test the sample to understand its function.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 95a68fb6-42c6-5a18-d72b-763f11131cf8
* Version Independent ID: 3d52e60a-f772-66e7-adf6-9ad79acdae0f
* Content: [Register-ArgumentCompleter (Microsoft.PowerShell.Core)](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/register-argumentcompleter?view=powershell-6)
* Content Source: [reference/6/Microsoft.PowerShell.Core/Register-ArgumentCompleter.md](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/Register-ArgumentCompleter.md)
* Product: **powershell**
* GitHub Login: @SteveL-MSFT
* Microsoft Alias: **slee**

### LINKED ISSUE #3604: CompletionResult class needs to be explained for Register-ArgumentCompleter
System.Management.Automation.CompletionResult should at least be linked to from this cmdlet and examples of parameters passed to it should be explained.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 95a68fb6-42c6-5a18-d72b-763f11131cf8
* Version Independent ID: 3d52e60a-f772-66e7-adf6-9ad79acdae0f
* Content: [Register-ArgumentCompleter (Microsoft.PowerShell.Core)](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/register-argumentcompleter?view=powershell-6)
* Content Source: [reference/6/Microsoft.PowerShell.Core/Register-ArgumentCompleter.md](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/Register-ArgumentCompleter.md)
* Product: **powershell**
* GitHub Login: @SteveL-MSFT
* Microsoft Alias: **slee**

### PR DESCRIPTION
**Pull request by bobbytreed**
Version(s) of document impacted
------------------------------
- [ ] Impacts 7 document
- [X] Impacts 6 document
- [X] Impacts 5.1 document
- [X] Impacts 5.0 document
- [ ] Impacts 4.0 document
- [ ] Impacts 3.0 document

Reason(s) for not updating all version of documents
--------------------------------------------------
- [ ] The documented feature was introduced in version (list version here) of PowerShell
- [ ] This issue only shows up in version (list version(s) here) of the document
- [ ] This PR partially fixes the issue, and issue #<insert here> tracks the remaining work

### CODE REVIEW

**Comment by zjalexander** [comment]
Docs Build status updates of commit _[feeceb5](https://github.com/bobbytreed/PowerShell-Docs/pull/4321/commits/feeceb57a829b34e6472610761cc0224769c0642)_: 

:clock10: **Full build**: the average full build time is _13 min(s) 55 sec(s)_, this is based on the last _17_ build(s) for this repository.

**Comment by zjalexander** [comment]
Docs Build status updates of commit _[feeceb5](https://github.com/bobbytreed/PowerShell-Docs/pull/4321/commits/feeceb57a829b34e6472610761cc0224769c0642)_: 

### :warning: Validation status: warnings


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[reference/5.0/Microsoft.PowerShell.Core/About/about_Functions_Advanced_Parameters.md](https://github.com/bobbytreed/PowerShell-Docs/blob/bobbytreed-issue3604/reference/5.0/Microsoft.PowerShell.Core/About/about_Functions_Advanced_Parameters.md) | :warning:Warning | [View](https://review.docs.microsoft.com/en-us/PowerShell/module/microsoft.powershell.core/about/about_functions_advanced_parameters?view=powershell-5.0&branch=pr-en-us-4321) (powershell-5.0) | [Details](#user-content-08ada56c-07c1f0b0)
[reference/5.1/Microsoft.PowerShell.Core/About/about_Functions_Advanced_Parameters.md](https://github.com/bobbytreed/PowerShell-Docs/blob/bobbytreed-issue3604/reference/5.1/Microsoft.PowerShell.Core/About/about_Functions_Advanced_Parameters.md) | :warning:Warning | [View](https://review.docs.microsoft.com/en-us/PowerShell/module/microsoft.powershell.core/about/about_functions_advanced_parameters?view=powershell-5.1&branch=pr-en-us-4321) (powershell-5.1) | [Details](#user-content-275cb513-07c1f0b0)
[reference/6/Microsoft.PowerShell.Core/About/about_Functions_Advanced_Parameters.md](https://github.com/bobbytreed/PowerShell-Docs/blob/bobbytreed-issue3604/reference/6/Microsoft.PowerShell.Core/About/about_Functions_Advanced_Parameters.md) | :warning:Warning | [View](https://review.docs.microsoft.com/en-us/PowerShell/module/microsoft.powershell.core/about/about_functions_advanced_parameters?view=powershell-6&branch=pr-en-us-4321) (powershell-6) | [Details](#user-content-692e398b-07c1f0b0)
[reference/5.0/Microsoft.PowerShell.Core/Register-ArgumentCompleter.md](https://github.com/bobbytreed/PowerShell-Docs/blob/bobbytreed-issue3604/reference/5.0/Microsoft.PowerShell.Core/Register-ArgumentCompleter.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/PowerShell/module/microsoft.powershell.core/register-argumentcompleter?view=powershell-5.0&branch=pr-en-us-4321) (powershell-5.0) |
[reference/5.1/Microsoft.PowerShell.Core/Register-ArgumentCompleter.md](https://github.com/bobbytreed/PowerShell-Docs/blob/bobbytreed-issue3604/reference/5.1/Microsoft.PowerShell.Core/Register-ArgumentCompleter.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/PowerShell/module/microsoft.powershell.core/register-argumentcompleter?view=powershell-5.1&branch=pr-en-us-4321) (powershell-5.1) |
[reference/6/Microsoft.PowerShell.Core/Register-ArgumentCompleter.md](https://github.com/bobbytreed/PowerShell-Docs/blob/bobbytreed-issue3604/reference/6/Microsoft.PowerShell.Core/Register-ArgumentCompleter.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/PowerShell/module/microsoft.powershell.core/register-argumentcompleter?view=powershell-6&branch=pr-en-us-4321) (powershell-6) |

<a id="08ada56c-07c1f0b0"></a>
### [reference/5.0/Microsoft.PowerShell.Core/About/about_Functions_Advanced_Parameters.md](https://github.com/bobbytreed/PowerShell-Docs/blob/bobbytreed-issue3604/reference/5.0/Microsoft.PowerShell.Core/About/about_Functions_Advanced_Parameters.md)
 - **[Warning]** `````Metadata with following name(s) are not allowed to be set in Yaml header, or as file level metadata in docfx.json, or as global metadata in docfx.json: locale. They are generated by Docs platform, so the values set in these 3 places will be ignored. Please remove them from all 3 places to resolve the warning.`````

<a id="275cb513-07c1f0b0"></a>
### [reference/5.1/Microsoft.PowerShell.Core/About/about_Functions_Advanced_Parameters.md](https://github.com/bobbytreed/PowerShell-Docs/blob/bobbytreed-issue3604/reference/5.1/Microsoft.PowerShell.Core/About/about_Functions_Advanced_Parameters.md)
 - **[Warning]** `````Metadata with following name(s) are not allowed to be set in Yaml header, or as file level metadata in docfx.json, or as global metadata in docfx.json: locale. They are generated by Docs platform, so the values set in these 3 places will be ignored. Please remove them from all 3 places to resolve the warning.`````

<a id="692e398b-07c1f0b0"></a>
### [reference/6/Microsoft.PowerShell.Core/About/about_Functions_Advanced_Parameters.md](https://github.com/bobbytreed/PowerShell-Docs/blob/bobbytreed-issue3604/reference/6/Microsoft.PowerShell.Core/About/about_Functions_Advanced_Parameters.md)
 - **[Warning]** `````Metadata with following name(s) are not allowed to be set in Yaml header, or as file level metadata in docfx.json, or as global metadata in docfx.json: locale. They are generated by Docs platform, so the values set in these 3 places will be ignored. Please remove them from all 3 places to resolve the warning.`````

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C5%5C20%5C6ff7e8c3-dfc6-3ebd-da5a-d5e2ff43de8f%5CPullRequest%5C201905201652126757-4321%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=EmuFd58dBp%2BimwhYkxe67mVaWIZGjnIk1M1Q0wdqRMo%3D&st=2019-05-20T17%3A01%3A15Z&se=2019-06-20T17%3A06%3A15Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

File: reference/5.0/Microsoft.PowerShell.Core/Register-ArgumentCompleter.md [hunk 1]
```diff
@@ -113,9 +191,42 @@ Accept wildcard characters: False
...
+  provided before they pressed `<TAB>`. Your script block should use this value
+  to determine tab completion values.
+- `$commandAst` (Position 3) - This parameter is set to the Abstract Syntax
+  Tree (AST) for the current input line. For more information, see
+  [Ast Class](/dotnet/api/system.management.automation.language.ast).
+- `$fakeBoundParameter` (Position 4) - This parameter is set to a hashtable
+  containing the `$PSBoundParameters` for the cmdlet, before the user pressed
+  `<TAB>`. For more information, see [about_Automatic_Variables](./About/about_Automatic_Variables.md).
+
+When you specify the **Native** parameter, the script block should take the following parameters in
```
Comment (rkeithhill) [inline reference/5.0/Microsoft.PowerShell.Core/Register-ArgumentCompleter.md]: My experience is that you also get the following parameters when you don't specify the `-ParameterName` parameter (even if you haven't specified -Native).

Comment (sdwheeler) [reply reference/5.0/Microsoft.PowerShell.Core/Register-ArgumentCompleter.md]: @rkeithhill So is there something that should be changed in the docs? How would you word it?

Comment (rkeithhill) [reply reference/5.0/Microsoft.PowerShell.Core/Register-ArgumentCompleter.md]: It is not when you specify `-Native`, it is when you don't specify `-ParameterName`.  Register-ArgumentCompleter has these two parameter sets:
```
Register-ArgumentCompleter -CommandName <string[]> -ScriptBlock <scriptblock> [-Native] [<CommonParameters>]

Register-ArgumentCompleter -ParameterName <string> -ScriptBlock <scriptblock> [-CommandName <string[]>] [<CommonParameters>]
```
If you specify `-ParameterName` you get the second parameter set and five parameters get sent to the ScriptBlock.  If you do not specify `-ParameterName`, then you get the first parameter set and you only get three parameters.  

A quick experiment verifies this:
```
function Test-Command([CmdletBinding][Parameter()][string]$Name) {$Name} 
$argCompleter = {$OFS=", "; $global:COMPLETER = "$($args | % GetType | Out-String)"}
Register-ArgumentCompleter -CommandName Test-Command -ScriptBlock $argCompleter
Test-Command <tab>
```
Now inspect the variable the scriptblock created:
```
PS> $COMPLETER                                                                                    
IsPublic IsSerial Name                                     BaseType
-------- -------- ----                                     --------
True     True     String                                   System.Object
True     False    CommandAst                               System.Management.Automation.Language.CommandBaseAst       
True     True     Int32                                    System.ValueType
```
So long story short, you could simply change line 216 to say `When you do not specify the **ParameterName** parameter, ...`
